core/web/resolver/auth.go

```diff
@@ -3,16 +3,53 @@ package resolver
 import (
 	"context"
 
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/sessions"
 	"github.com/smartcontractkit/chainlink/core/web/auth"
 )
 
-// Authenticates the user from the session cookie.
+// Authenticates the user from the session cookie, presence of user inherently provides 'view' access.
 func authenticateUser(ctx context.Context) error {
-	_, ok := auth.GetGQLAuthenticatedSession(ctx)
+	if _, ok := auth.GetGQLAuthenticatedSession(ctx); !ok {
+		return unauthorizedError{}
+	}
+	return nil
+}
+
+// Authenticates the user from the session cookie and asserts at least 'run' role.
+func authenticateUserCanRun(ctx context.Context) error {
+	session, ok := auth.GetGQLAuthenticatedSession(ctx)
 	if !ok {
 		return unauthorizedError{}
 	}
+	if session.User.Role == sessions.UserRoleView {
+		return errors.Errorf("Not permitted with current role %s\n", session.User.Role)
```

Comment:
Maybe consider a standard error for these?
```suggestion
		return RoleNotPermittedErr(session.User.Role)
```
```go
type RoleNotPermittedErr sessions.UserRole

func (r RoleNotPermittedErr) Error() string { return fmt.Sprintf("not permitted with current role: %s", r) }
```

Reply:
Sure 🙂  Factored out into suggested `RoleNotPermittedErr`